cuckoo/table.go

```diff
@@ -77,22 +38,22 @@ type Table struct {
 // numBuckets = number of buckets
 // depth = the number of entries per bucket
 // randSeed = seed for PRNG
-func NewTable(name string, numBuckets int, bucketDepth int, itemSize int,
+func NewTable(name string, numBuckets uint64, bucketDepth uint64, itemSize uint64,
 	data []byte, randSeed int64) *Table {
 	t := &Table{name, numBuckets, bucketDepth, itemSize, nil, nil, nil, nil}
-	t.log = log.New(os.Stderr, "[Cuckoo:"+name+"] ", log.Ldate|log.Ltime|log.Lshortfile)
-	t.rand = rand.New(rand.NewSource(randSeed))
-
 	if data == nil {
 		data = make([]byte, numBuckets*bucketDepth*itemSize)
 	}
-	if len(data) != numBuckets*bucketDepth*itemSize {
-		return nil
-	}
 	t.data = data
-
+	t.rand = rand.New(rand.NewSource(randSeed))
```

Comment:
with coordinator, we no longer need table behavior to be deterministic, and can prefer crypto/rand instead, correct?